_articles/custom-mouse-buttons.md

```diff
@@ -1,20 +1,51 @@
 ---
 layout: article
-title: Setting Custom Mouse Actions
+title: Customizing Your Mouse
 description: >
-   How to set custom mouse actions with xbindkeys
+   How to customize the buttons on your mouse
 keywords:
   - Support
   - Pop
   - Ubuntu
   - xbindkeys
+  - Solaar
+  - Piper
 image: http://support.system76.com/images/system76.png
 hidden: false
 section: software-applications
 
 ---
 
-### Software
+### Graphical Software
+
+#### Piper
+
+![Piper Buttons Window](/images/custom-mouse-buttons/Piper-2.png) 
+
+<u>Piper</u> is a frontend to the <u>libratbag</u> backend for setting custom actions to mouse buttons like the Logitech MX Master 3 mouse that we sell and this section will be using that mouse as an example for using <u>Piper</u>. You can see the source code of <u>Piper</u> [here](https://github.com/libratbag/piper).
+
+![Piper Buttons Assign Window](/images/custom-mouse-buttons/Piper-3.png)
+
+In this window the button that has been clicked can be reassigned to a differet function like changing the Left-click to the Right-click or swapping the Forward button to the Back button for web pages.
+
+#### Solaar
+
+![Solaar Main Window](/images/custom-mouse-buttons/Solaar.png) 
+
+![Solaar Main Window with information](/images/custom-mouse-buttons/Solaar-2.png) 
+
+<u>Solaar</u> is a graphical tool to for adding more devices to your Logitech receiver and information like the following:
```

Comment:
is a graphical tool to for adding more devices to your Logitech receiver and information like the following:

Change to:
is a graphical tool for adding more devices to your Logitech USB Unify receiver. It also provides information on the following:

Reply:
This is also still a run-on/incompatible sentence. 

right now it reads: adding more devices to your receiver and [adding more devices to your] information.